core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -2955,16 +2955,17 @@ public boolean test(Project project) {
 
   @Test public void testPullAggregateThroughUnion() {
     HepProgram program = new HepProgramBuilder()
+        .addRuleInstance(AggregateProjectMergeRule.INSTANCE)
         .addRuleInstance(AggregateUnionAggregateRule.INSTANCE)
         .build();
 
-    final String sql = "select deptno, job from"
-        + " (select deptno, job from emp as e1"
-        + " group by deptno,job"
+    final String sql = "select job, deptno from"
+        + " (select job, deptno from emp as e1"
```

Comment:
Would you please add a new test rather than update existing one?

Reply:
Thanks. I have addressed your comment and updated the PR.